encode_test.go

```diff
@@ -148,3 +148,25 @@ func TestEncoding(t *testing.T) {
 		}
 	}
 }
+
+func TestEncoding_Date_NotExact(t *testing.T) {
```

Comment:
What's the reasoning behind not handling these with the other test cases?

Reply:
Its annoying but if I put it in the same block, then it will fail on the decoded test will fail

Reply:
Is that only until you fix the decode?

Reply:
Makes sense, thanks for the explanation!